test/jdk/java/foreign/NativeTestHelper.java

```diff
@@ -123,6 +123,25 @@ public static boolean isPointer(MemoryLayout layout) {
      */
     public static final AddressLayout C_POINTER = ValueLayout.ADDRESS
             .withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, C_CHAR));
+    /**
+     * The layout for the {@code size_t} C type
+     */
+    public static final ValueLayout C_SIZE_T = ValueLayout.ADDRESS.byteSize() == 8 ? C_LONG_LONG : C_INT;
+
+    // Common layout shared by some tests
+    // struct S_PDI { void* p0; double p1; int p2; };
+    public static final MemoryLayout S_PDI_LAYOUT = switch ((int) ValueLayout.ADDRESS.byteSize()) {
+        case 8 -> MemoryLayout.structLayout(
+            C_POINTER.withName("p0"),
+            C_DOUBLE.withName("p1"),
+            C_INT.withName("p2"),
+            MemoryLayout.paddingLayout(4));
+        case 4 -> MemoryLayout.structLayout(
+            C_POINTER.withName("p0"),
+            C_DOUBLE.withName("p1"),
+            C_INT.withName("p2"));
+        default -> throw new UnsupportedOperationException("Unsupported address size");
+    };
```

Comment:
Both TestUpcallHighArity and TestUpcallStructScope used this layout. I've moved it here to avoid declaring it twice.